enes-homework-week3-strings -and-arrays.js

```diff
@@ -0,0 +1,36 @@
+//Q1
+
+let myString = 'hello,this,is,a,difficult,to,read,sentence';
+console.log("myString: the result will be 'hello,this,is,a,difficult,to,read,sentence'");
+console.log(myString);
+console.log('the lenght of myString will be 42');
+console.log(myString.length);
+
+let myNewString = myString.split(',').join(' ');
+
+myString = myNewString;
+
+console.log("myString: the result will be 'hello this is a difficult to read sentence'");
+console.log(myString);
+
+//Q2
+
+let favoriteAnimals = ['blowfish', 'capricorn', 'giraffe'];
+favoriteAnimals.push('turtle');
+console.log('favoriteAnimals : the result will be: blowfish, capricorn, giraffe, turtle');
+console.log(favoriteAnimals);
+
+favoriteAnimals.splice(1, 0, 'meerkat');
+console.log(favoriteAnimals); //expected output: blowfish meerkat capricorn giraffe turtle
+console.log(favoriteAnimals.length); //expected output: 5
```

Comment:
This is correct but you will learn a little later that using `splice` actually changes the `favoriteAnimals` array itself. This is called mutating the data structure and often it is suggested that we don't mutate the data structure i.e try to push the `meerkat` in between the two elements but return a new array itself thus having two arrays at the end, original and the modified array.

Maybe when you are introduced to this concept, you can revisit it and think of a way by which you can still add `meerkat` but don't modify the existing array. 🙂

Reply:
I have found the solution. But i still don't know what type of problems/bugs that a mutation can do.
here is the solution:  
const newArr = favoriteAnimals.splice(0,2).concat("meerkat").concat(favoriteAnimals(2,6); is this true?

Reply:
Ah. Not quite there yet. Because you are still using `splice` method which would mutate the `favoriteAnimals` array again. Hint: try exploring `slice` method instead and see if you can somehow make it work that way.